File: Golang/CountAndSay/CountAndSay_test.go
package CountAndSay

import "testing"

func TestCountAndSay(t *testing.T) {

	if countAndSay(1) == "1" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if countAndSay(2) == "11" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if countAndSay(3) == "21" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if countAndSay(4) == "1211" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if countAndSay(5) == "111221" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if countAndSay(6) == "312211" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/CountAndSayTest.py
from unittest import TestCase

from CountAndSay import CountAndSay


class TestCountAndSay(TestCase):
    def test_countAndSay(self):
        cas = CountAndSay()

        self.assertTrue(cas.countAndSay(1) == "1")

        self.assertTrue(cas.countAndSay(2) == "11")

        self.assertTrue(cas.countAndSay(3) == "21")

        self.assertTrue(cas.countAndSay(4) == "1211")

        self.assertTrue(cas.countAndSay(5) == "111221")

        self.assertTrue(cas.countAndSay(6) == "312211")
